fdbserver/ConsistencyScan.actor.cpp

```diff
@@ -319,7 +320,19 @@ ACTOR Future<int> consistencyCheckReadData(Database cx,
 					bool isExpectedTSSMismatch = g_network->isSimulated() &&
 					                             g_simulator->tssMode == ISimulator::TSSMode::EnabledDropMutations &&
 					                             isTss;
-					TraceEvent(isExpectedTSSMismatch ? SevWarn : SevError, "ConsistencyCheck_DataInconsistent")
+					// It's possible that the storage servers are inconsistent in KillRegion
+					// workload where a forced recovery is performed. The killed storage server
+					// in the killed region returned first with a higher version, and a later
+					// response from a different region returned with a lower version (because
+					// of the rollback of the forced recovery). In this case, we should not fail
+					// the test. So we double check both process are live.
+					bool isFailed =
+					    g_network->isSimulated() &&
+					    (g_simulator->getProcessByAddress((*storageServerInterfaces)[j].address())->failed ||
```

Comment:
Shouldn't this also check that both processes are in different regions as a tighter check? Otherwise this could ignore legitimate inconsistencies in the non-multi-region case

Reply:
Good callout! Done.